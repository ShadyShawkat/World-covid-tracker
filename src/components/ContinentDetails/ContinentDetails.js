/* eslint-disable implicit-arrow-linebreak, function-paren-newline */
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import CountUp from 'react-countup';
import images from '../../images';

import Spinner from '../Spinner/Spinner';
import CountryList from '../CountryList/CountryList';

const ContinentDetails = () => {
  const { continentName } = useParams();
  const continentDetails = useSelector((state) =>
    state.covid.continents.find((cont) => cont.continent === continentName),
  );
  const countries = useSelector((state) =>
    state.covid.countries.filter(
      (country) => country.continent === continentName,
    ),
  );

  return (
    <>
      {!continentDetails && <Spinner />}
      {continentDetails && (
        <>
          <div className="flex md:flex-row flex-col gap-6 md:gap-3 items-center w-full justify-center pb-16 pt-6 px-3 bg-blue-dark rounded-t-lg">
            <img
              src={images[continentName]}
              alt="continent map"
              className="w-40 invert"
            />
            <div className="flex flex-col font-bold text-2xl text-white">
              <CountUp
                delay={1}
                end={continentDetails.cases}
                separator=","
                prefix="Total Cases: "
                duration={3}
              />
              <CountUp
                delay={1}
                prefix="Total Deaths: "
                end={continentDetails.deaths}
                separator=","
                duration={3}
              />
            </div>
          </div>
          <CountryList countries={countries} />
        </>
      )}
    </>
  );
};

export default ContinentDetails;
