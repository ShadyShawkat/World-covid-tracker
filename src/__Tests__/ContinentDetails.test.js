import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Router, { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import ContinentDetails from '../components/ContinentDetails/ContinentDetails';

const intialState = {
  continents: [
    { continent: 'Asia', cases: 11111 },
    { continent: 'Africa', cases: 22222 },
    { continent: 'Europe', cases: 33333 },
    { continent: 'Australia-Oceania', cases: 44444 },
    { continent: 'North America', cases: 55555 },
    { continent: 'South America', cases: 66666 },
  ],
  countries: [
    {
      country: 'Egypt',
      cases: '1111',
      continent: 'Africa',
      countryInfo: { flag: '' },
    },
    {
      country: 'India',
      cases: '2222',
      continent: 'Asia',
      countryInfo: { flag: '' },
    },
    {
      country: 'Indonesia',
      cases: '3333',
      continent: 'Asia',
      countryInfo: { flag: '' },
    },
    {
      country: 'Canada',
      cases: '4444',
      continent: 'North America',
      countryInfo: { flag: '' },
    },
    {
      country: 'Algeria',
      cases: '5555',
      continent: 'Africa',
      countryInfo: { flag: '' },
    },
  ],
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));

describe('Home component', () => {
  test('Test Rendering countries', () => {
    const mockStore = configureStore([thunk]);
    const store = mockStore({ covid: intialState });

    jest
      .spyOn(Router, 'useParams')
      .mockReturnValue({ continentName: 'Africa' });

    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <ContinentDetails />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
    const countryItem = screen.getAllByTestId('countryItem');
    expect(countryItem).toHaveLength(2);
  });
});
