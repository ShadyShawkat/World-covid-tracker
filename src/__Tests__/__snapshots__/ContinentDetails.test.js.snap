// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Test Rendering countries 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex md:flex-row flex-col gap-6 md:gap-3 items-center w-full justify-center pb-16 pt-6 px-3 bg-blue-dark rounded-t-lg"
      >
        <img
          alt="continent map"
          class="w-40 invert"
          src="Africa.png"
        />
        <div
          class="flex flex-col font-bold text-2xl text-white"
        >
          <span
            class="text-center md:text-left mb-4 text-4xl"
          >
            Africa
          </span>
          <span>
            Total Cases: 0
          </span>
          <span>
            Total Deaths: 0
          </span>
        </div>
      </div>
      <ul
        class="grid md:grid-cols-2 rounded-lg overflow-hidden gap-0.5"
      >
        <li
          class="country-card"
        >
          <div
            class="flex flex-col px-3 py-10 bg-no-repeat relative"
            data-testid="countryItem"
            style="background-size: 50% 100%;"
          >
            <div
              class="flex flex-col gap-2 self-end font-bold text-white items-end"
            >
              <h4
                class="flex flex-col items-end text-right"
              >
                <span
                  class="text-2xl"
                >
                  Egypt
                </span>
              </h4>
              <span
                class="text-xl"
              >
                1111
              </span>
            </div>
          </div>
        </li>
        <li
          class="country-card"
        >
          <div
            class="flex flex-col px-3 py-10 bg-no-repeat relative"
            data-testid="countryItem"
            style="background-size: 50% 100%;"
          >
            <div
              class="flex flex-col gap-2 self-end font-bold text-white items-end"
            >
              <h4
                class="flex flex-col items-end text-right"
              >
                <span
                  class="text-2xl"
                >
                  Algeria
                </span>
              </h4>
              <span
                class="text-xl"
              >
                5555
              </span>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex md:flex-row flex-col gap-6 md:gap-3 items-center w-full justify-center pb-16 pt-6 px-3 bg-blue-dark rounded-t-lg"
    >
      <img
        alt="continent map"
        class="w-40 invert"
        src="Africa.png"
      />
      <div
        class="flex flex-col font-bold text-2xl text-white"
      >
        <span
          class="text-center md:text-left mb-4 text-4xl"
        >
          Africa
        </span>
        <span>
          Total Cases: 0
        </span>
        <span>
          Total Deaths: 0
        </span>
      </div>
    </div>
    <ul
      class="grid md:grid-cols-2 rounded-lg overflow-hidden gap-0.5"
    >
      <li
        class="country-card"
      >
        <div
          class="flex flex-col px-3 py-10 bg-no-repeat relative"
          data-testid="countryItem"
          style="background-size: 50% 100%;"
        >
          <div
            class="flex flex-col gap-2 self-end font-bold text-white items-end"
          >
            <h4
              class="flex flex-col items-end text-right"
            >
              <span
                class="text-2xl"
              >
                Egypt
              </span>
            </h4>
            <span
              class="text-xl"
            >
              1111
            </span>
          </div>
        </div>
      </li>
      <li
        class="country-card"
      >
        <div
          class="flex flex-col px-3 py-10 bg-no-repeat relative"
          data-testid="countryItem"
          style="background-size: 50% 100%;"
        >
          <div
            class="flex flex-col gap-2 self-end font-bold text-white items-end"
          >
            <h4
              class="flex flex-col items-end text-right"
            >
              <span
                class="text-2xl"
              >
                Algeria
              </span>
            </h4>
            <span
              class="text-xl"
            >
              5555
            </span>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
